src/steam_network/websocket_client.py

```diff
@@ -127,8 +137,14 @@ async def run(self, create_future_factory: Callable[[], Future]=asyncio_future):
                 )
                 await sleep(RECONNECT_INTERVAL_SECONDS)
                 continue
+            except AuthenticationRequired:
+                logger.error("Authentication lost mid use. Restarting the socket, auth, and run loops")
+                #Interface checks if user name is info cache and raises an authentication required if it's note there. 
+                #Clearing the cache here will result in that error being raised, which lets gog know to redo auth.
```

Comment:
```suggestion
                #Interface checks if user name is in info cache and raises an AuthenticationRequired if it's not there. 
                #Clearing the cache here will result in that error being raised, which lets gog know to redo auth.
```

> Interface

which interface? in Galaxy API or internal protocol_client interface? 

> that error being raised

Did you mean the same error `AuthenticationRequired` being raised from `plugin.authenticate` method?

Reply:
I believe the comment meant the interface between gog and our backend code, but that definitely needs cleaning up. The first thing every function gog function checks (friends, games, etc) is I'd the user info cache is valid. If it's not, it raises AuthenticationRequired (from `galaxy.api.exceptions` iirc). This causes gog to trigger "connection lost. Reconnect" or equivalent. 

I'll clean that up.

Reply:
added comments around the whole while loop and clarified the comments here